_resources/port1.0/group/crossgdb-1.0.tcl

```diff
@@ -0,0 +1,179 @@
+                gdb/doc     gdb/doc/Makefile.in     stabs       texinfo
+            }
+
+            foreach { path makefile name suffix } ${infopages} {
+                set src      ${worksrcpath}/${path}/${name}.${suffix}
+                set makefile ${worksrcpath}/${makefile}
+                # If the makefile doesn't exists, skip it.
+                if { ! [file exists ${makefile}] } {continue}
+
+                # Fix the source
+                reinplace -q "s|setfilename ${name}.info|setfilename ${crossgdb.target}-${name}.info|g" ${src}
+                reinplace -q "s|(${name})|(${crossgdb.target}-${name})|g" ${src}
+                reinplace -q "s|@file{${name}}|@file{${crossgdb.target}-${name}}|g" ${src}
+
+                # Rename the source
+                file rename ${worksrcpath}/${path}/${name}.${suffix} \
+                    ${worksrcpath}/${path}/${crossgdb.target}-${name}.${suffix}
+
+                # Fix the Makefile
+                reinplace -q -E "s:\[\[:<:\]\]${name}\\.(info|pod|${suffix}):${crossgdb.target}-&:g" ${makefile}
+
+                # Fix install-info's dir.
+                # (note: this may be effectless if there was no info dir to be fixed)
+                reinplace -q "s|--info-dir=\$(DESTDIR)\$(infodir)|--dir-file=\$(DESTDIR)\$(infodir)/${crossgdb.target}-gdb-dir|g" ${makefile}
+            }
+
+            # Do not install libiberty
+            reinplace -q {/^install:/s/ .*//} ${worksrcpath}/libiberty/Makefile.in
+        }
+
+        # Older Apple clang versions segfault, c.f.,
+        # https://trac.macports.org/ticket/56883
+        compiler.blacklist  {clang < 601}
+        compiler.thread_local_storage yes
+
+        # Needs C++11; halfway redundant due to the blacklist above, but make
+        # sure selected compiler supports the standard - getting rid of old
+        # Apple GCC versions and the like?
+        compiler.cxx_standard 2011
+
+        compiler.blacklist  {clang < 601}
+
+        configure.dir       ${workpath}/build
+        configure.cmd       ${worksrcpath}/configure
+        configure.args      \
+                            --target=${crossgdb.target} \
+                            --infodir=${prefix}/share/info \
+                            --mandir=${prefix}/share/man \
+                            --with-docdir=${prefix}/share/doc \
+                            --with-gdb-datadir=${prefix}/share/${name} \
+                            --without-guile \
+                            --without-python \
+                            --disable-werror
+
+        pre-configure {
+            file mkdir ${configure.dir}
+        }
+
+        build.dir ${configure.dir}
+
+        # this port installs files to ${prefix}/${crossgdb.target}
+        destroot.violate_mtree yes
+
+        post-destroot {
+            # Avoid conflicts with ${crossgdb.target}-binutils port
+            delete ${destroot}${prefix}/share/info/${crossgdb.target}-bfd.info
```

Comment:
this should probably use the Tcl `file delete`?

Reply:
Is it better not to use macros in PortGroup?

Reply:
The standard TCL `file xxx` commands are fine, but the MacPorts-provided macros are generally preferable:

https://guide.macports.org/#reference.tcl-extensions